File: packages/jupyter-scales/src/log.ts
// Copyright (c) Jupyter Development Team.
// Distributed under the terms of the Modified BSD License.

import {
  WidgetModel, ManagerBase
} from '@jupyter-widgets/base';

import {
  scaleLog
} from 'd3-scale';

import {
  JUPYTER_EXTENSION_VERSION
} from './version';

import {
  ContinuousScaleModel
} from './continuous';



/**
 * A widget model of a linear scale
 */
export
class LogScaleModel extends ContinuousScaleModel {
  defaults() {
    return {...super.defaults(),
      _model_name: LogScaleModel.model_name,
    };
  }

  /**
   * Create the wrapped d3-scale scaleLinear object
   */
  constructObject(): any {
    return scaleLog();
  }

  static serializers = {
      ...ContinuousScaleModel.serializers,
    }

  static model_name = 'LogScaleModel';
}

File: ipyscales/log.py
#!/usr/bin/env python
# coding: utf-8

# Copyright (c) Jupyter Development Team.
# Distributed under the terms of the Modified BSD License.

"""
Defines logarithmic scale widget, and any supporting functions
"""

from traitlets import Float, Unicode, Union, Bool

from .scale import Scale
from .traittypes import FullColor, VarlenTuple

class LogScale(Scale):
    """A logarithmic scale widget.

    See the documentation for d3-scale's scaleLog for
    further details.
    """
    _model_name = Unicode('LogScaleModel').tag(sync=True)

    domain = VarlenTuple(trait=Float(), default_value=(0., 1.), minlen=2).tag(sync=True)
    range = VarlenTuple(trait=Union((Float(), FullColor())), default_value=(0., 1.), minlen=2).tag(sync=True)

    interpolator = Unicode('interpolate').tag(sync=True)
    clamp = Bool(False).tag(sync=True)

    base = Float(10).tag(sync=True)
